File: Python3/Minimum Moves to Equal Array Elements.py
# 72ms, 71.51%
class Solution:
    def minMoves(self, nums):
        min_num, res = nums[0], 0
        for n in nums:
            if n < min_num: min_num = n
            res += n - nums[0]
        return res + (nums[0] - min_num) * len(nums)
    

File: Cpp/Minimum Moves to Equal Array Elements.cpp
// 51ms, 90.20%
class Solution {
public:
    int minMoves(vector<int> &nums) {
        int min = nums[0], res = 0;
        for (auto n : nums) {
            if (n < min) min = n;
            res += (n - nums[0]);
        }
        return res + (nums[0] - min) * nums.size();
    }
};
